azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -606,6 +912,112 @@ func flattenEventGridEventSubscriptionSubjectFilter(filter *eventgrid.EventSubsc
 	return []interface{}{result}
 }
 
+func flattenEventGridEventSubscriptionAdvancedFilter(input *[]eventgrid.BasicAdvancedFilter) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	boolEquals := make([]interface{}, 0)
+	numberGreaterThan := make([]interface{}, 0)
+	numberGreaterThanOrEquals := make([]interface{}, 0)
+	numberLessThan := make([]interface{}, 0)
+	numberLessThanOrEquals := make([]interface{}, 0)
+	numberIn := make([]interface{}, 0)
+	numberNotIn := make([]interface{}, 0)
+	stringBeginsWith := make([]interface{}, 0)
+	stringEndsWith := make([]interface{}, 0)
+	stringContains := make([]interface{}, 0)
+	stringIn := make([]interface{}, 0)
+	stringNotIn := make([]interface{}, 0)
+
+	for _, item := range *input {
+		switch f := item.(type) {
+		case eventgrid.BoolEqualsAdvancedFilter:
+			boolEquals = append(boolEquals, map[string]interface{}{
```

Comment:
which means this can become:

```suggestion
			boolEquals = append(boolEquals, flattenValues(f.Key, f.Value))
```

(I guess we could also pass in `f`)